Add train/test stats to model reports (#504)
ccao-data/model-res-avm: reports/performance/_train_test.qmd

# Triad-Level Machine Learning Metrics

This section compares model performance metrics at the triad level between the
**training set** (the data the model learned from) and the **test set** (a
held-out sample the model never saw during training). Large gaps between train
and test metrics — particularly for RMSE, MDAPE, or MAE — may indicate
overfitting, where the model has memorized the training data rather than
learning generalizable patterns. The sale price and prediction distributions
are included to confirm that both splits are drawn from similar underlying
price ranges.

```{r _train_test_setup_script}
source("../_setup.R")

> if (!exists("model_performance_train")) {

Reviewer: Do we want to put this in `setup` like the other downloads?

Reviewer 2: Go for it.

  model_performance_train <-
    arrow::read_parquet(paths$output$performance_train$local)
}
```

```{r}
test_stats <- model_performance_test %>%
  filter(

>     geography_type == "triad_code",

Reviewer 2: It might be good to note that the model isn't just trained on one triad, so these metrics need to be digested in that context if we're using them to evaluate over-fitting.

Author: Good call.  Will clarify.

Author: Updated the section heading to containt "**Note** onsistent with the rest of this report, we report these splits at the triad level, even though the model was trained on county-wide data.  We do this, as we are most concerned with whether or not we have generalized appropriately (not-overfit) at the triad-level."

    triad_code == run_triad_code,
    by_class == FALSE
  ) %>%
  select(
    rmse, mdape, mae, starts_with("sale_fmv_"),
    starts_with("estimate_fmv_")
  )

train_stats <- model_performance_train %>%
  filter(
    geography_type == "triad_code",
    triad_code == run_triad_code,
    by_class == FALSE
  ) %>%
  select(
    rmse, mdape, mae, starts_with("sale_fmv_"),
    starts_with("estimate_fmv_")
  )

train_test_comparison_tbl <- dplyr::bind_rows(list(
  "train" = train_stats, "test" = test_stats
), .id = "split")
```

```{r}
train_test_comparison_tbl %>%
  select(
    split, rmse, mdape, mae,
    sale_fmv_mean, sale_fmv_sd,
    estimate_fmv_mean, estimate_fmv_sd
  ) %>%
  kable(
    digits = 3,
    format.args = list(big.mark = ","),
    caption = "Train vs. Test Set Performance Metrics"
  )

> ```

Reviewer 2: Can we format these values with dollar signs like in the Topline Statistics table? Likewise, can we clean up the column names so they're readable/more in line with column names in the rest of the report?

Author: will do

Author: done

```{r}
train_test_comparison_tbl %>%
  select(
    split,
    sale_fmv_min, sale_fmv_q25, sale_fmv_median, sale_fmv_q75, sale_fmv_max
  ) %>%
  kable(
    digits = 3,
    format.args = list(big.mark = ","),
    caption = "Actual Sale Price Distributions, for Reference"
  )
```

```{r}
train_test_comparison_tbl %>%
  select(
    split,
    estimate_fmv_min, estimate_fmv_q25, estimate_fmv_median,
    estimate_fmv_q75, estimate_fmv_max
  ) %>%
  kable(
    digits = 3,
    format.args = list(big.mark = ","),

>     caption = "Predicted Value Distributions, for Reference"

Reviewer 2: Do we need "for Reference" here? I'm not sure what else this data would be for.

# Triad-Level Machine Learning Metrics

This section compares model performance metrics at the triad level between the
**training set** (the data the model learned from) and the **test set** (a
held-out sample the model never saw during training). Large gaps between train
and test metrics — particularly for RMSE, MDAPE, or MAE — may indicate
overfitting, where the model has memorized the training data rather than
learning generalizable patterns. The sale price and prediction distributions
are included to confirm that both splits are drawn from similar underlying
price ranges.

> **Note**: Consistent with the rest of this report, we report these splits at the triad level, even though the model was trained on county-wide data.  We do this, as we are most concerned with whether or not we have generalized appropriately (not-overfit) at the triad-level.

Reviewer 2: ```suggestion
**Note**: Consistent with the rest of this report, while the model was trained on county-wide data, we report these splits at the triad level since that is where we are most concerned with overfitting.
```

```{r _train_test_setup_script}
source("../_setup.R")
```

> ```{r}

Reviewer: Can we name the code chunks here like we do in the other report scripts?

{r _sales_volume_town_year}

Author: done
